### ISSUE TITLE
Lambda logic assertions

### ISSUE THREAD

**Issue by goreRatzete** [MarkusAmshove/Kluent#77]
Just had some use cases where I was missing some infix assertions. 
In my example I basically want to assert that a List<String> contains some String which is not blank. 

My initial solution was this:
```
val specs: List<String>
specs.filter(String::isNotBlank).shouldNotBeEmpty()
```

```
specs should { any(String::isNotBlank) }
specs shouldContainSomeWhere String::isNotBlank
```

The implementation looks like this:
```
infix inline fun <T> T.should(logic: T.() -> Boolean) = if (logic(this)) Unit else throw AssertionError()
infix inline fun <T> List<T>.shouldContainSomeWhere(logic: T.() -> Boolean) = if (any(logic)) Unit else throw AssertionError()
```
The cool thing is that you can pass function references 👍 

There are many more applications for this. 
What do you think?

**Comment by MarkusAmshove** [comment]
I really like the idea. We could allow to pass an optional string as failure message, that way everyone can write a small wrapper for business-logik assertions :-)

**Comment by MarkusAmshove** [comment]
I've just released v1.32 which contains these changes 👍
